tests/components/providers/test_training_tracker_provider.py

```diff
@@ -0,0 +1,55 @@
+from typing import Dict, Text, Any
+
+import pytest
+from rasa.components.providers.training_tracker_provider import TrainingTrackerProvider
+from rasa.engine.graph import ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import Domain
+from rasa.shared.core.generator import TrackerWithCachedStates
+from rasa.shared.core.training_data.story_reader.yaml_story_reader import (
+    YAMLStoryReader,
+)
+from rasa.shared.core.training_data.structures import StoryGraph
+
+
+@pytest.mark.parametrize(
+    "config, expected_trackers",
+    [
+        ({}, 507),
+        ({"augmentation_factor": 0}, 7),
+        ({"use_story_concatenation": False}, 7),
```

Comment:
The `expected_trackers` output is really a test of the `TrainingDataGenerator` (and not so much of any functionality inside the TrainingTrackerProvider). I would expect these to be tests in `TrainingDataGenerator`. ... Not really easy to test what is done (only) here then. We could mock the TrainingDataGenerator and test that the constructor and generate is used 😅  . Maybe another reason to make `TrainingDataGenerator` a graph component. Wdyt?